pkg/transformers/selectbypath.go

```diff
@@ -0,0 +1,51 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package transformers
+
+import "k8s.io/apimachinery/pkg/runtime/schema"
+
+// selectByPath returns true if path `p` matches any `GroupVersionKind` and `Path` in `pathConfigs`.
```

Comment:
This is ripe for optimization.  However, as long as we treat zero values as wildcards and add `GroupVersionKind` entries in `skipNamespacePathConfigs`, average runtime complexity should remain fairly low.